pam_duo/authlogin_duo.te

```diff
@@ -1,9 +1,10 @@
-module authlogin_duo 1.0;
+module authlogin_duo 1.0.1;
 
 require {
     type http_port_t;
+    type http_cache_port_t;
```

Comment:
If this is `required` my understanding is this type must be defined for the semodule command to succeed. Do we have any guarantees this will exist on all systems even if they aren't using it? I tested it on my centos7 box and it was there, but I was wondering if you knew any places that document it officially. I couldn't find any in my googling.

Reply:
> Is there any reason an admin wouldn't just wanna add the one new port to http_port_t like this? Seems like it would be the more restrictive option to just allow the one you need.
> 
> ```
> semanage port -a -t http_port_t -p tcp 8080
> ```

Because ports can only have a single label, and 8080 already has the `http_cache_port_t` label:

```
$ sudo semanage port -a -t http_port_t -p tcp 8080
ValueError: Port tcp/8080 already defined

```

Even if it were possible to apply the `http_port_t` label to port 8080, this would require making a customization to the SELinux policy for every host which will have the duo_unix package installed. The entire point of having a package contribute its own SELinux policy is specifically to prevent administrators from having to add custom overrides on a per-host basis.

Also, FWIW, I've argued in the past that the distinction between `http_port_t` and `http_cache_port_t` is dumb, and should be removed in favor of a boolean that indicates whether the system uses a proxy:

https://bugzilla.redhat.com/show_bug.cgi?id=1305469#c8

> If this is required my understanding is this type must be defined for the semodule command to succeed. Do we have any guarantees this will exist on all systems even if they aren't using it? I tested it on my centos7 box and it was there, but I was wondering if you knew any places that document it officially. I couldn't find any in my googling.

The `http_cache_port_t` label has been present in selinux core policy since at least 2008:

https://github.com/SELinuxProject/selinux/blob/13cd4c8960688af11ad23b4c946149015c80d549/libsepol/tests/policies/test-cond/refpolicy-base.conf

It's been present in Red Hat Enterprise Linux since at least RHEL5:

https://bugzilla.redhat.com/show_bug.cgi?id=234126

It's present in all current (non-EOL) versions of Fedora (26 and 27).

Thus, I think you're very unlikely to find an SELinux-enabled distribution that supports the `http_port_t` label but not the `http_cache_port_t` label.

You can test this in Travis, BTW. If one or more labels are unknown in the current policy, the semodule command will fail to load the module. Here's what happens if I modify my patch to instead permit connections to the `http_cachee_port_t` (note the typo) label:

```
$ make
…
$ sudo semodule -n -s targeted -X 200 -i pam_duo/authlogin_duo.pp 
Failed to resolve typeattributeset statement at /etc/selinux/targeted/tmp/modules/200/authlogin_duo/cil:2
semodule:  Failed!
```

The only potential wrinkle is that the `semodule` command has to be run as root. So if Travis can't run commands as root while testing a build, Travis won't be able to test this.

Reply:
Great thanks for all the explanation there. I believe that as long as RHEL5 has the feature and sufficiently old fedoras systems do (I found references all the way back to 20) we should be okay.
I'll look into adding this to our Travis, but we also have a system that will try our build on a bunch of different OSes so hopefully we'd catch it there too.
Overall this change looks good to me so I'll get a second reviewer from Duo and we'll get this merged in. Thanks again for the PR!